File: code/generate_strategies/IR(pseudocode)/java/java/function_decimal_infinite.java
package com.example;

import org.apache.commons.lang3.math.NumberUtils;

public class InfiniteCheck {
    public String check_discount_for_large_order(String value) {
        try {
            if (value.equals("Infinity") || value.equals("-Infinity")) {
                return "True";
            }
            
            if (value.equals("NaN")) {
                return "False";
            }
            
            if (!NumberUtils.isCreatable(value)) {
                return "Error";
            }
            return "False";
        } catch (Exception e) {
            return "Error";
        }
    }
}

File: code/generate_strategies/IR(pseudocode)/cpp/cpp/function_decimal_infinite.cpp
#include <string>
#include <boost/multiprecision/cpp_dec_float.hpp>

class InfiniteCheck {
public:
    std::string check_discount_for_large_order(const std::string& value) {
        try {
            
            if (value == "Infinity" || value == "+Infinity" || value == "-Infinity") {
                return "True";
            }
            
            
            boost::multiprecision::cpp_dec_float_50 decimal_value(value);
            
            
            if (isinf(decimal_value)) {
                return "True";
            }
            return "False";
        }
        catch (const std::exception&) {
            if (value == "NaN") {
                return "False";
            }
            return "Error";
        }
    }
};